### PR TITLE
feature/table-sorting-by-click

### PR DESCRIPTION
**Pull request by Konmly**
Sort the table by clicking on the buttons.

table

### CODE REVIEW

File: Table-sorting-by-click/test.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      };
+      default:
+        break;
+  }
+
+  arr.sort(compare);
+  tbody.append(...arr);
+}
+
+document.addEventListener("click", function (event) {
```
Comment (PavelTankovich) [inline Table-sorting-by-click/test.js]: Prefer arrow func

File: Table-sorting-by-click/test.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  if (event.target.tagName != "TH") {
+    return;
+  }
+
+  const e = event.target;
+
+  sortTable(e.cellIndex, e.dataset.type, "grid");
```
Comment (PavelTankovich) [inline Table-sorting-by-click/test.js]: this func has only 2 arguments

File: Table-sorting-by-click/test.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+function sortTable(colNum, type) {
+  const tbody = table.querySelector("tbody");
+
+  const arr = Array.from(tbody.rows);
+
+  let compare;
+
+  switch (type) {
+    case "number":
+      compare = function (rowA, rowB) {
```
Comment (PavelTankovich) [inline Table-sorting-by-click/test.js]: arrow func

File: Table-sorting-by-click/test.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  arr.sort(compare);
+  tbody.append(...arr);
+}
+
+document.addEventListener("click", function (event) {
+  if (event.target.tagName != "TH") {
+    return;
+  }
+
+  const e = event.target;
```
Comment (PavelTankovich) [inline Table-sorting-by-click/test.js]:
```suggestion
  const target = event.target;
```

Comment (PavelTankovich) [reply Table-sorting-by-click/test.js]: and move it up

File: Table-sorting-by-click/test.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+const table = document.querySelector(".grid");
+
+function sortTable(colNum, type) {
+  const tbody = table.querySelector("tbody");
+
+  const arr = Array.from(tbody.rows);
+
+  let compare;
```
Comment (PavelTankovich) [inline Table-sorting-by-click/test.js]: why is it called "compare"?

File: Table-sorting-by-click/test.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        return rowA.cells[colNum].innerHTML > rowB.cells[colNum].innerHTML
+          ? 1
+          : -1;
+      };
+      default:
+        break;
+  }
+
+  arr.sort(compare);
+  tbody.append(...arr);
```
Comment (PavelTankovich) [inline Table-sorting-by-click/test.js]: Have the previous rows been removed?

**Review by PavelTankovich — CHANGES_REQUESTED** [review]

File: Table-sorting-by-click/test.js (hunk 5 above)
Comment (Konmly) [reply Table-sorting-by-click/test.js]: Next we compare rows to sort, would it be better to call it sorting?

File: Table-sorting-by-click/test.js (hunk 6 above)
Comment (Konmly) [reply Table-sorting-by-click/test.js]: no

Comment (Konmly) [reply Table-sorting-by-click/test.js]: no

**Review by PavelTankovich — APPROVED** [review]

**Pull request merged**